pkg/migration-tool/utils/client.go

```diff
@@ -0,0 +1,239 @@
+package utils
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/opentracing-contrib/go-stdlib/nethttp"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/config"
+	configutil "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/version"
+	"github.com/prometheus/prometheus/prompb"
+)
+
+var userAgent = fmt.Sprintf("Prometheus/%s", version.Version)
+
+const maxErrMsgLen = 512
+
+type Client struct {
+	remoteName string
+	url        *config.URL
+	Client     *http.Client
+	timeout    time.Duration
+}
+
+// clientConfig configures a client.
+type clientConfig struct {
+	URL              *configutil.URL
+	Timeout          model.Duration
+	HTTPClientConfig configutil.HTTPClientConfig
+}
+
+// NewClient creates a new read or write client. The `clientType` should be either `read` or `write`.
+func NewClient(remoteName, clientType string, conf *clientConfig) (*Client, error) {
+	httpClient, err := configutil.NewClientFromConfig(conf.HTTPClientConfig, fmt.Sprintf("remote_storage_%s_client", clientType), false, false)
+	if err != nil {
+		return nil, err
+	}
+
+	t := httpClient.Transport
+	httpClient.Transport = &nethttp.Transport{
+		RoundTripper: t,
+	}
+
+	return &Client{
+		remoteName: remoteName,
+		url:        conf.URL,
+		Client:     httpClient,
+		timeout:    time.Duration(conf.Timeout),
+	}, nil
+}
+
+// Read reads from a remote endpoint. It returns the response size of compressed and uncompressed in bytes.
+func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryResult, int, int, error) {
```

Comment:
can we give names to return arguments so easier to read what's happening from declaration